source/slang/slang-ir-simplify-cfg.cpp

```diff
@@ -444,6 +444,30 @@ static bool trySimplifyIfElse(IRBuilder& builder, IRIfElse* ifElseInst)
             return true;
         }
     }
+    else
+    {
+        // Otherwise, we can try to remove at least remove one of the trivial branches
+        // Remove either the true or false block if it jumps to the after block
+        // with no parameters.
+
+        const auto afterBlock = ifElseInst->getAfterBlock();
+        if(!afterBlock->getFirstParam())
+        {
+            const auto trueBlock = ifElseInst->getTrueBlock();
+            const auto falseBlock = ifElseInst->getFalseBlock();
+
+            if(isTrueBranchTrivial && trueBlock != afterBlock && !trueBlock->hasMoreThanOneUse())
+            {
+                trueBlock->replaceUsesWith(afterBlock);
+                trueBlock->removeAndDeallocate();
+            }
+            else if(isFalseBranchTrivial && falseBlock != afterBlock && !falseBlock->hasMoreThanOneUse())
+            {
+                falseBlock->replaceUsesWith(afterBlock);
+                falseBlock->removeAndDeallocate();
+            }
+        }
+    }
```

Comment:
@csyonghe This was quite simple, but wasn't done before; is there a reason for that? Making this change did necessitate a fix here: https://github.com/shader-slang/slang/pull/2899/files#diff-88028dd102858f855a13f83c516f5bc3d10d434d43e0c9ecb5b12305b7194e60R196-R203

Reply:
I think I removed something similar to this to get some autodiff test to pass. Maybe that associated fix is the reason.